.github/workflows/test.yml

```diff
@@ -212,6 +212,15 @@ jobs:
         env:
           RUST_BACKTRACE: 'full'
 
+      - name: Check for Rust file changes
+        id: check-rust-changes
+        run: |
+          if [[ -n "$(git diff origin/${{ github.base_ref }} HEAD -- ./rust)" ]]; then
```

Comment:
this means even agent-config changes will lead to a full sealevel e2e, do we want that?

Reply:
it's hard not to have false positives but it's better than false negatives imo - cargo.toml and .lock files could change, and possibly other configs too

Reply:
since most agent config changes happen on new deploy batches, I think it's acceptable

Reply:
sgtm!